app/core/nice/quest.py

```diff
@@ -218,7 +224,13 @@ async def get_nice_quest_phase_no_rayshift(
     }
 
     if raw_quest.mstQuestPhaseDetail:
+        mstSpot = await war.get_spot_from_id(conn, raw_quest.mstQuestPhaseDetail.spotId)
```

Comment:
Can you add a check if the mstQuestPhaseDetail.spotId is different from mstQuest.id and only fetch the spot from the db if necessary?